packages/shared/src/components/cards/PostCardHeader.tsx

```diff
@@ -51,6 +56,15 @@ export const PostCardHeader = ({
     bookmarked: post.bookmarked && !showFeedback,
   });
 
+  const shouldUseInternalLink =
+    isSharedPostSquadPost(post) || isInternalReadType(post.sharedPost);
+  const sharedArticleLink =
+    shouldUseInternalLink &&
+    (shouldUseInternalLink
+      ? post.sharedPost?.commentsPermalink
+      : post.sharedPost?.permalink);
+  const articleLink = post.sharedPost ? sharedArticleLink : postLink;
```

Comment:
```suggestion
  const articleLink = useMemo(() => {
    if (post.sharedPost) {
      const shouldUseInternalLink =
        isSharedPostSquadPost(post) || isInternalReadType(post.sharedPost);

      return shouldUseInternalLink
        ? post.sharedPost?.commentsPermalink
        : post.sharedPost?.permalink;
    }

    return postLink;
  }, [post, postLink]);
```

maybe more readable?

Reply:
memo is here just for the sake of computation being more readable, it also could just be inline code with let or helper function